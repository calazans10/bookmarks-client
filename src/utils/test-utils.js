import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import rootReducer from '../reducers';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
