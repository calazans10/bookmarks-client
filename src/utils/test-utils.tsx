import React, { ReactNode } from 'react';
import { AnyAction, Action, createStore, Store } from 'redux';
import { Provider } from 'react-redux';
import { render, RenderResult } from '@testing-library/react';
import rootReducer from '../reducers';

interface RenderWithRedux<
  S = any,
  A extends Action = AnyAction,
  I extends S = any
  > {
  (
    ui: ReactNode,
    reduxOptions?: {
      store?: Store<S, A>
      initialState?: I
    }
  ): RenderResult & {
    store: Store<S, A>
  }
}

export const renderWithRedux: RenderWithRedux = (
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
