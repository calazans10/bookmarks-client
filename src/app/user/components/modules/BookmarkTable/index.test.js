import React from 'react';
import { renderWithRedux } from 'test-utils';
import { BookmarkTable } from './index';

describe('BookmarkTable', () => {
  let modalRoot = document.getElementById('modal-root');
  if (!modalRoot) {
    modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  }

  it('renders without crashing', () => {
    const { container } = renderWithRedux(
      <BookmarkTable
        bookmarks={[
          {
            id: '9b2bfb9a-3776-48ca-835a-2c17ccef44c6',
            url: 'https://reactjs.org/blog/2017/12/07/introducing-the-react-rfc-process.html',
            title: 'Introducing the React RFC Process',
            user: {
              id: 'e4f262c4-8dd3-4db4-85c8-83e03b8ecad4',
              name: 'John Doe',
              email: 'john.doe@example.com',
              is_admin: false,
            },
          },
        ]}
        onChangeLocation={jest.fn()}
        onShowConfirm={jest.fn()}
        onChangeSelectedBookmark={jest.fn()}
        onRequestDeleteBookmark={jest.fn()}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
