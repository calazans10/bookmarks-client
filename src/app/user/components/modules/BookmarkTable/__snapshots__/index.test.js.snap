// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkTable renders without crashing 1`] = `
<table
  class="sc-AxjAm doFumh"
>
  <tbody>
    <tr
      class="sc-AxirZ dPZPVA"
    >
      <td
        class="sc-AxiKw cyHeZF"
        data-label="Title"
      >
        <a
          href="https://reactjs.org/blog/2017/12/07/introducing-the-react-rfc-process.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Introducing the React RFC Process
        </a>
      </td>
      <td
        class="sc-AxiKw dnGOKB"
        data-label="Actions"
      >
        <button
          class="sc-AxhCb bzqWB"
          type="button"
        >
          Edit
        </button>
        <button
          class="sc-AxhCb iPDvrL"
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
